src/mcts/search.cc

```diff
@@ -318,8 +318,43 @@ void Search::MaybeTriggerStop() {
   // Don't stop when the root node is not yet expanded.
   if (total_playouts_ == 0) return;
 
+  bool kldgain_too_small = false;
```

Comment:
Maybe factor out to a separate function? For other conditions it's just nice 5 lines.
Also this function is called from all search threads and watchdog thread, and it sets some search global variables. Maybe there is a better place to do that?

(this function is kind of intended for polling at random times).

Reply:
Extraction done.  It is really safe to call anytime - but now its only called by search threads.
Added another global variable to track whether to exit.  Also marked all the global variables as guarded by counters lock.